src/ray/core_worker/core_worker_process.cc

```diff
@@ -151,6 +147,13 @@ CoreWorkerProcessImpl::~CoreWorkerProcessImpl() {
   if (options_.enable_logging) {
     RayLog::ShutDownRayLog();
   }
+
+  {
+    absl::WriterMutexLock lock(&mutex_);
+    if (core_worker_ != nullptr) {
+      core_worker_ = nullptr;
```

Comment:
Why add this section? The reset of `core_worker_` is already done in `ShutdownDriver` and `RunWorkerTaskExecutionLoop`.

Reply:
Good catch.